lib/services/unified_resource.go

```diff
@@ -392,7 +392,12 @@ func makeResourceSortKey(resource types.Resource) resourceSortKey {
 	case types.AppServer:
 		app := r.GetApp()
 		if app != nil {
-			name = app.GetName()
+			friendlyName := types.FriendlyName(app)
+			if friendlyName != "" {
+				name = friendlyName
+			} else {
+				name = app.GetName()
+			}
```

Comment:
Just FYI - Once we're on Go 1.22 on release branches this will be expressed more cleanly as 
```
name := cmp.Or(types.FriendlyName(..), app.GetName())
```